src/python/OrientedBox_TEST.py

```diff
@@ -0,0 +1,310 @@
+# Copyright (C) 2021 Open Source Robotics Foundation
+#
+# Licensed under the Apache License, Version 2.0 (the "License")
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#       http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+from ignition.math import IGN_PI, OrientedBoxd, MassMatrix3d, Material, Pose3d, Vector3d
+
+g_tolerance = 1e-6
+
+
+class TestOrientedBox(unittest.TestCase):
+
+    def test_empty_constructor_new(self):
+        box = OrientedBoxd()
+
+        self.assertEqual(box.size(), Vector3d.ZERO)
+        self.assertEqual(box.pose(), Pose3d.ZERO)
+        # self.assertEqual(Material(), box.material())
+
+    def test_size_only_constructor(self):
+        box = OrientedBoxd(Vector3d(1, 2, 3))
+        self.assertEqual(box.size(), Vector3d(1, 2, 3))
+        self.assertEqual(box.pose(), Pose3d.ZERO)
+
+    def test_negative_size_constructor(self):
+        box = OrientedBoxd(Vector3d(-1, 0, -3))
+        self.assertEqual(box.size(), Vector3d(1, 0, 3))
+
+    def test_size_pose_constructor(self):
+        box = OrientedBoxd(Vector3d(1, 2, 3), Pose3d(-1, -2, -3, 0, 1, 2))
+        self.assertEqual(box.size(), Vector3d(1, 2, 3))
+        self.assertEqual(box.pose(), Pose3d(-1, -2, -3, 0, 1, 2))
+
+    def test_copy_constructor(self):
+
+        box1 = OrientedBoxd(Vector3d(0.1, 0.2, 0.3),
+                            Pose3d(-0.1, -0.2, 0.0, 1.1, 1.2, 1.3))
+        box2 = OrientedBoxd(box1)
+
+        self.assertEqual(box2.size(), Vector3d(0.1, 0.2, 0.3))
+        self.assertEqual(box2.pose(), Pose3d(-0.1, -0.2, 0.0, 1.1, 1.2, 1.3))
+
+    def test_length(self):
+        box = OrientedBoxd(Vector3d(0.1, -2.1, 0.0))
+        self.assertEqual(box.xlength(), 0.1)
+        self.assertEqual(box.ylength(), 2.1)
+        self.assertEqual(box.zlength(), 0.0)
```

Comment:
I know we've already merged support for `AxisAlignedBox` with `XLength()` in c++ mapped to `xlength()` in python (and simlar for y, z), but I think it would look nicer as `x_length`, `y_length` and `z_length`. Is there an easy way to do that?

Reply:
oh, I see that we've done that for Quaternion and PID already with `%rename()`, so I'll add that here

Reply:
https://github.com/ignitionrobotics/ign-math/pull/276/commits/fc260dbe4f3785119db9ebdbaa3807e025e9e449

Reply:
similar change for `AxisAlignedBox` in https://github.com/ignitionrobotics/ign-math/pull/281